packages/rest-client-jsonapi/src/services/utils/fetch/fetch.ts

```diff
@@ -22,8 +22,8 @@ export async function fetchJsonApi<T>(
   })
 
   if (!response.ok) {
-    // todo proper validation
-    throw Error("request failed")
+    const json = await response.json()
+    return Promise.reject(json?.errors || "Unknown error")
```

Comment:
I wasn't sure if this should return a `new Error(string)` or just `string`... I don't think we need stacktrace for this, this is handled and not throwing by default, the actual network call was successful and we're just reporting the reason data was not returned. open to opinions here though 😄